### PR TITLE
Tech/allow many args CH-789

### PR DESCRIPTION
**Pull request by ndelangen**
https://linear.app/chromaui/issue/CH-789/chromatic-cli-should-accept-multiples-of-the-same-argument-and-pick

@zol I think this should fix the issue..

### CODE REVIEW

File: bin-src/lib/getOptions.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default async function getOptions({ argv, env, flags, log, packageJson })
     originalArgv: argv,
 
     buildScriptName: flags.buildScriptName,
-    outputDir: flags.outputDir,
+    outputDir: Array.isArray(flags.outputDir) ? takeLast(flags.outputDir) : undefined,
```
Comment (tmeasday) [inline bin-src/lib/getOptions.js:53]: Doesn't `takeLast` already include this ternary?

**Review by tmeasday — CHANGES_REQUESTED** [review]
Small tweak maybe?

File: bin-src/lib/getOptions.js (hunk 1 above)
Comment (ndelangen) [reply bin-src/lib/getOptions.js:53]: The value should always be an array of strings (thanks to `isMultiple: true`), but if it isn't -somehow- I want to return `undefined`.

Comment (tmeasday) [reply bin-src/lib/getOptions.js:53]: image

Are you concerned about the possibility of the flag not being an array but also not being `undefined`? Is that actually possible with the library? Would that concern not apply to other options we are `takeLast`-ing? (In which case should `takeLast` just return `undefined` in the non-array case?)

**Comment by zol** [comment]
Progress here looks good. Sadly I don't have the time/expertise to do code reviews anymore so I've taken myself off the list here.

**Comment by ghengeveld** [comment]
@ndelangen What's the status of this? Perhaps you'll want to revisit something based on @tmeasday's comment?

**Comment by ndelangen** [comment]
Very strange for the tests to start failing all of a sudden after merging in `main`? @ghengeveld got any ideas?

**Comment by ghengeveld** [comment]
Sorry, my bad.

**Comment by ghengeveld** [comment]
I don't think this PR is going to completely address the problem. For example, in [this](https://app.intercom.com/a/apps/zj7sn9j1/inbox/inbox/conversation/27253970011#part_id=comment-27253970011-11751369752) example, the `--webpack-stats-json` flag is getting passed to `build-storybook` twice, and getting rejected as a result. We could tweak the Storybook CLI to deal with that, but really Chromatic should prevent the same flag from getting set twice when it invokes `build-storybook`. I think we should resolve [`buildScriptName`](https://github.com/chromaui/chromatic-cli/blob/3e4c4d3d8b59fd5db61df96b800d610cb0b03b72/bin-src/tasks/build.js#L57) to the underlying command and strip `--output-dir` and `--webpack-stats-json` (and their alias / camelCase form) from that command before invoking it.

**Comment by ndelangen** [comment]
I see, this PR is still an improvement I guess, a step into the right direction?

**Comment by ndelangen** [comment]
@ghengeveld could you add your review on this?

File: bin-src/main.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ export async function runBuild(ctx) {
     ctx.options = await getOptions(ctx);
   } catch (e) {
     ctx.log.info('');
-    ctx.log.error(e.message);
+    ctx.log.error(e.stack || e.message);
```
Comment (ghengeveld) [inline bin-src/main.js:73]: This should be wrapped in `fatalError` like below. Just throwing a stack trace at the user isn't good UX.

**Review by ghengeveld — APPROVED** [review]

**Pull request merged**